### PR TITLE
Quick search improvements

### PR DESCRIPTION
**Pull request by irahov**
A number of changes:

Add capability to project content into the quick search
Always show section name in the quick search
Add data property to the search providers
Fix bug that manually selected item is not preserved in async case

Please take a look at the individual comments

### CODE REVIEW

File: projects/components/src/quick-search/quick-search.component.ts [hunk 1]
```diff
@@ -198,7 +200,9 @@ export class QuickSearchComponent {
             }
             searchSection.result = searchResult;
             searchSection.isLoading = false;
-            this.selectFirst(true);
+            if (this.selectedItem == null) {
```
Comment (juanmendes) [inline projects/components/src/quick-search/quick-search.component.ts:203]: We usually prefer `=== null` unless you are trying to specifically test for undefined also

File: projects/components/src/quick-search/quick-search.service.spec.ts [hunk 2]
```diff
@@ -4,11 +4,15 @@
...
-import { QuickSearchProvider } from './quick-search.provider';
+import { QuickSearchProviderDefaults } from './quick-search.provider';
 import { QuickSearchService } from './quick-search.service';
 
-class SimpleSearchProvider implements QuickSearchProvider {
-    constructor(public sectionName: string = '', public order: number = -1) {}
+class SimpleSearchProvider extends QuickSearchProviderDefaults {
+    constructor(sectionName: string = '', order: number = -1) {
+        super();
+        this.sectionName = sectionName;
```
Comment (juanmendes) [inline projects/components/src/quick-search/quick-search.service.spec.ts:13]: Why was this changed, previous syntax is shorter

File: projects/examples/src/components/quick-search/quick-search-sync-async.example.component.html [hunk 3]
```diff
@@ -16,6 +16,21 @@
         description="The text to be shown in the quick search input field when it is empty"
     >
     </vcd-form-input>
+    <vcd-form-checkbox
```
Comment (juanmendes) [inline projects/examples/src/components/quick-search/quick-search-sync-async.example.component.html:19]: Ideally, we'd have an example for this feature, not have this embedded in an example that is meant to show async results

**Review by juanmendes — APPROVED** [review]

File: projects/examples/src/components/quick-search/quick-search-sync-async.example.component.html (hunk 3 above)
Comment (irahov) [reply projects/examples/src/components/quick-search/quick-search-sync-async.example.component.html:19]: reverted
